packages/storybook-utils/src/source-code-generator.ts

```diff
@@ -373,21 +384,55 @@ const generateSlotChildrenSourceCode = (
           (param) => !["{", "}"].includes(param),
         );
 
-        const parameters = paramNames.reduce<Record<string, string>>((obj, param) => {
-          obj[param] = `{{ ${param} }}`;
-          return obj;
-        }, {});
+        const parameters: Record<string, string> = {};
+        const proxied: Record<string, Record<string, unknown>> = {};
+        paramNames.forEach((param) => {
+          parameters[param] = `{{ ${param} }}`;
+          proxied[param] = new Proxy(
```

Comment:
I think some explanatory comments could here since its not self explanatory in my opinion :)